github-runners/terraform/base/main.tf

```diff
@@ -0,0 +1,380 @@
+    ]
+
+    on_failure = continue
+
+  }
+
+  # Setting up the ssh connection to install the runner server
+  connection {
+    type        = "ssh"
+    host        = element(self.*.public_ip, count.index)
+    user        = "ec2-user"
+    private_key = file("${var.public_key_path}")
+    timeout     = "90s"
+  }
+
+}
+
+
+resource "aws_instance" "mac_arm64_runner" {
+  count             = length(var.public_subnet_cidrs)
+  ami               = data.aws_ami.mac_arm64.image_id
+  instance_type     = element(var.instance_type_mac, 1)
+  subnet_id         = element(aws_subnet.public_subnets.*.id, count.index)
+  availability_zone = element(var.azs, count.index)
+  # Security Group
+  vpc_security_group_ids = ["${aws_security_group.allow_runner.id}"]
+  # the Public SSH key
+  key_name                    = var.aws_key_name
+  associate_public_ip_address = true
+  tenancy                     = "host"
+
+  tags = {
+    name       = "gh-macos-arm64-runner"
+    role       = "runner"
+    os_name    = "macos"
+    os_version = "12"
+    os_name    = "Monterey"
+    arch       = "arm64"
+  }
+
+  depends_on = [
+    aws_subnet.public_subnets,
+    aws_nat_gateway.nat_gateway,
+    aws_internet_gateway.gw
+  ]
+
+  lifecycle {
+
+    create_before_destroy = true
+
+  }
+
+  provisioner "file" {
+    source      = "./scripts/secure.sh"
+    destination = "/home/ec2-user/secure.sh"
+  }
+
+  provisioner "file" {
+    source      = "./scripts/cleanup.sh"
+    destination = "/home/ec2-user/cleanup.sh"
+  }
+
+  # Github runner installation
+  provisioner "remote-exec" {
+    inline = [
+      "sleep 160",
+      "mkdir actions-runner && cd actions-runner",
+      "curl -o actions-runner-osx-x64-${var.gh_runner_version}.tar.gz -L https://github.com/actions/runner/releases/download/v${var.gh_runner_version}/actions-runner-osx-arm64-${var.gh_runner_version}.tar.gz",
+      "echo '${lookup(var.gh_runner_checksums, "mac_arm64", "")}  actions-runner-osx-arm64-${var.gh_runner_version}.tar.gz' | shasum -a 256 -c",
+      "tar xzf ./actions-runner-osx-arm64-${var.gh_runner_version}.tar.gz",
+      "./config.sh --url https://github.com/subspace --token ${var.gh_token} --unattended --name mac_arm64 --labels 'self-hosted,MacOS,ARM64' --work _work --runasservice",
+      "sudo ./run.sh",
+      "curl https://my-netdata.io/kickstart.sh > /tmp/netdata-kickstart.sh && sh /tmp/netdata-kickstart.sh --non-interactive --nightly-channel --claim-rooms ${var.netdata_room}  --claim-token ${var.netdata_token} --claim-url https://app.netdata.cloud",
+    ]
+
+    on_failure = continue
+
+  }
+
+  # Setting up the ssh connection to install the runner server
+  connection {
+    type        = "ssh"
+    host        = element(self.*.public_ip, count.index)
+    user        = "ec2-user"
+    private_key = file("${var.public_key_path}")
+    timeout     = "90s"
+  }
+
+}
+
+
+resource "aws_instance" "windows_x86_64_runner" {
```

Comment:
any plan to caputure the windows metrics through netdata. I see they mentioned an option here - https://blog.netdata.cloud/how-to-monitor-windows-systems-with-netdata/

Reply:
It's not supported yet in dashboard